bgpd/bgp_evpn_vty.c

```diff
@@ -2550,7 +2551,8 @@ static void evpn_show_route_vni_macip(struct vty *vty, struct bgp *bgp,
 		return;
 	}
 
-	build_evpn_type2_prefix(&p, mac ? mac : &empty_mac, ip);
+	build_evpn_type2_prefix(&p, mac ? mac : &empty_mac,
```

Comment:
Isn't better to handle all this NULL pointer cases inside build_evpn_type2_prefix?

Reply:
But seems `build_evpn_type2_prefix()` only does simple things, needn't check pointers of `ipaddr` and `ethaddr` NULL or not. So always the callers ( like this commit, it is related to show command )  do these checks.